bundles/org.openhab.binding.bluetooth.bluegiga/src/main/java/org/openhab/binding/bluetooth/bluegiga/handler/BlueGigaBridgeHandler.java

```diff
@@ -476,18 +525,28 @@ public boolean bgConnect(BluetoothAddress address, BluetoothAddressType addressT
      * @return
      */
     public boolean bgDisconnect(int connectionHandle) {
+        logger.debug("BlueGiga Disconnect: connection {}", connectionHandle);
+        passiveScanIdleTimer.cancel();
         BlueGigaDisconnectCommand command = new BlueGigaDisconnectCommand();
         command.setConnection(connectionHandle);
-        BlueGigaDisconnectResponse response = (BlueGigaDisconnectResponse) getBgHandler().sendTransaction(command);
 
-        return response.getResult() == BgApiResponse.SUCCESS;
+        try {
+            BlueGigaDisconnectResponse response = getBgHandler().sendTransaction(command,
+                    BlueGigaDisconnectResponse.class, COMMAND_TIMEOUT_MS);
+            return response.getResult() == BgApiResponse.SUCCESS;
+        } catch (TimeoutException e) {
+            logger.debug("Timeout occured when sending disconnect command to device {}.", address);
+            return false;
+        } finally {
+            passiveScanIdleTimer.schedule(configuration.passiveScanIdleTime);
+        }
     }
 
     /**
      * Device discovered. This simply passes the discover information to the discovery service for processing.
      */
     public void deviceDiscovered(BluetoothDevice device) {
-        if (discoveryActive) {
+        if (configuration.discovery) {
```

Comment:
This doesn't take into account activeBackgroundDiscovery, where the user has manually activated a scan for devices to search for things. By always using the config the user will not be able to do a manual scan. You should keep the `discoveryActive` flag and just update it whenever a manual scan is requested through the `scanStart` and `scanStop` methods. You can use org.openhab.binding.bluetooth.bluez.handler.BlueZBridgeHandler for reference.

Reply:
Good point, but I didn't the change the behaviour, it has been always working like this